addons/account/models/chart_template.py

```diff
@@ -60,14 +62,12 @@ def _avoid_name_conflict():
                 for index, conflict_taxes in enumerate(conflict_taxes):
                     conflict_taxes.name = f"[old{index if index > 0 else ''}] {conflict_taxes.name}"
 
-        template_vals = template._get_tax_vals_complete(company)
-        chart_template = env['account.chart.template'].with_context(default_company_id=company.id)
         if old_tax:
             xml_id = old_tax.get_external_id().get(old_tax.id)
             if xml_id:
                 _remove_xml_id(xml_id)
         _avoid_name_conflict()
-        chart_template.create_record_with_xmlid(company, template, 'account.tax', template_vals)
+        template._generate_tax(company)
```

Comment:
@oco-odoo btw I don't know if you noticed this change but it is worth pinpoint it: `_generate_tax()` will create taxes in chart_template.country_id if there is one, else in the fiscal country.
It is necessary to be able to update taxes that don't belong to the company's main chart_template.

What this change implies is only for a very specific case that we have already faced so I think it's fine:
- I have a company with l10n_fr installed
- I use Guadeloupe as fiscal country
- My taxes (with l10n_fr's xmlid) are in Guadeloupe (because of 14.0 upgrade :( )
=> 1) If the update only provoke recreations of taxes (example: a change of rates) taxes will be created in France
=> 2) If the update has update of taxes (example: a change of tag) it will fail since it will try to add French tags to Guadeloupe taxes and rollback the whole transaction